### DISCUSSION TITLE
Define a Color Range - Use Case

### DISCUSSION THREAD

**Discussion by CarelessCourage** [jaames/iro.js#206] [Feature Requests]
### **Context**
Im working on a full color system for work. The color system I am developing is basically just a simple way to handle a color scheme in a way that ensures the colors are readable against each other no matter which colors the user picks. For instace, if they pick a dark background color then the system forces him to pick a light foreground color for the text. That way the text is visible. 

I was going to create a color picker for this system myself, using canvas. Thats how I came across Iro, and its geniunly the only color picker thats good enough that it has convinced me not to make my own custom one. The only thing thats missing for me is a simple way to limit a color selection to a certain lightness. 

### **Use case:** 
In my use case it would be very handy if I can give the user a visual way of saying "looking at your background color, here is an acceptable range of colors you can pick for your foreground colors." 

**Basically, I would like to show them a color picker that has all the available colors and values except the very dark once, or the very light once depending on some other condition.**

**Comment by sgrumo** [comment]
Hey @CarelessCourage, did you apply a workaround for your use case? I happen to have the same functional requirement and I wanted to know if you successfully had an idea.

**Reply by CarelessCourage** [reply]
No. Best I could come up with was hide parts of the color picker with parent overflow. Actually this issue made me build my own color picker. But its not really finished
